src/lib/operators/table_scan.cpp

```diff
@@ -43,7 +43,21 @@ namespace opossum {
 TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in,
                      const std::shared_ptr<AbstractExpression>& predicate)
     : AbstractReadOnlyOperator{OperatorType::TableScan, in, nullptr, std::make_unique<PerformanceData>()},
-      _predicate(predicate) {}
+      _predicate(predicate) {
+  /**
+   * Register as a consumer for all subplans of uncorrelated subqueries
+   */
+  auto pqp_subquery_expressions = collect_pqp_subquery_expressions(predicate);
+  for (const auto& subquery_expression : pqp_subquery_expressions) {
+    // We do not register for the subplans of correlated subqueries because they are templated and cannot be
+    // executed without concrete parameters. Thus, there is no option for result sharing at this point.
+    if (subquery_expression->is_correlated()) continue;
+
+    // Register & Store a pointer for easy deregistration later on.
+    subquery_expression->pqp->register_consumer();
```

Comment:
I just realized how important it is to do the registration as part of the constructor, not during execution. That should be documented in abstract_operator.hpp, including an example of what can go wrong if you do not do this.